### DISCUSSION TITLE
Autocomplete, Stub-files (.pyi) and More...

### DISCUSSION THREAD

**Discussion by BilakshanP** [PyO3/maturin#1805] [Q&A]
I am currently working on Python library using Maturin-PyO3 bindings.

For, myself I have created a stub-file in .env/lib/py_lib_name/py_lib_name.pyi, which helps me get the autocomplete results...

But in .gitignore file the venv names .env is mentioned, thus the stub-file I have written won't be helping out anybody who clones the repo from github, I was wondering how can I work-around this limitation.

I sure that I am doing something wrong, but it would be awesome if I could be nudged towards the correct path by somebody...

I myself am using vs-code on wsl.

**Comment by messense** [comment]
RTD: https://maturin.rs/project_layout#adding-python-type-information
